src/createRouter.tsx

```diff
@@ -210,9 +212,47 @@ export function createRouter(...args: any[]): UmbrellaRouter {
           unblock();
         };
       },
+      listen: (handler) => navigationHandlerManager.add(handler),
     },
   };
 
+  const routeContext = React.createContext<UmbrellaRoute | null>(null);
+
+  return router;
+
+  function Router(props: { children?: any }) {
```

Comment:
I'd use `React.ReactNode` instead of `any` here.

Reply:
I may end up doing just that. The goal here was to avoid a direct dependency on `@types/react`. I suppose it would likely be a peer dep but it feels like there's less a chance for conflicts with the exact version of `@types/react` Type Route is using if we just don't use it at all. Since TS is structurally typed this should work just fine. Still need to ponder a bit on if the reasoning holds up under closer scrutiny.